File: frontend/src/types/books.ts
export interface BookBaseSchema {
    user_id: string,
    author: string,
    title: string
}

export interface BookUploadResponseSchema extends BookBaseSchema {
    id: string,
    created_at: string,
    file_type: string,
    original_file_name: string,
    file_size: number,
    is_parsed: boolean
}

export interface BookResponseSchema extends BookBaseSchema {
    id: string,
    created_at: string,
    file_type: string,
    cover_image_base64?: string,
    is_parsed: boolean
}

export interface BookUpdateSchema {
    author?: string,
    title?: string
}

File: backend/schemas/books.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel
import uuid
from ..models.books import FileType


class BookBaseSchema(BaseModel):
    user_id: uuid.UUID
    author: str
    title: str


class BookUploadResponseSchema(BookBaseSchema):
    id: uuid.UUID
    upload_date: datetime
    file_type: FileType
    original_file_name: str
    file_size: int
    is_parsed: bool


class BookResponseSchema(BookBaseSchema):
    id: uuid.UUID
    upload_date: datetime
    file_type: FileType
    cover_image_base64: Optional[str] = None
    is_parsed: bool


class BookUpdateSchema(BaseModel):
    author: str = None
    title: str = None
